### ISSUE TITLE
 Setup fails with: PHP Notice:  A non well formed numeric value encountered inapp/PartKeeprRequirements.php on line 126

### ISSUE THREAD

**Issue by vogelfreiheit** [partkeepr/PartKeepr#980]
I have encountered this (blocking) issue in my installation. Setup runs until an error is encountered as follows:

https://i.imgur.com/PBFh10L.png

The response is JSON and apparently all tests complete successfully, but something in the backend processing seems to be expecting a different output.

# System Information

* PartKeepr Version: Git
* Reproducible on the demo system: No. Needs to run Setup.  
* Nginx configuration as documented in the Wiki.

# How to reproduce

Browse setup and run the initial tests.

**Comment by Drachenkaetzchen** [comment]
Please provide all details as per [CONTRIBUTING.md](https://github.com/partkeepr/PartKeepr/blob/master/CONTRIBUTING.md). No information about OS, PHP version etc given.

**Comment by blakangel** [comment]
I'm having the same issue as original poster. Fresh installation from git. Error occurs during the "Testing for correct PHP settings" part of the Prerequisites tab of the setup.

**OS**: FreeBSD 11.2-RELEASE-p8
**PHP**: 7.2.14
**Browser**: Chrome Version 72.0.3626.109 (Official Build) (64-bit)

**Nginx error log:** `2019/02/19 22:55:20 [error] 87258#101174: *3698 FastCGI sent in stderr: "PHP message: PHP Notice:  A non well formed numeric value encountered in /var/www/partkeepr/app/PartKeeprRequirements.php on line 126" while reading response header from upstream, client: xxxxx, server: xxxxx, request: "POST /setup/tests/check.php HTTP/1.1", upstream: "fastcgi://127.0.0.1:9000", host: "xxxxx", referrer: "http://xxxxxx/setup/index.html"`


**Chrome Javascript console**:
```
ext-all-debug.js:9392 [E] Ext.JSON.decode(): You're trying to decode an invalid JSON String: <br />
<b>Notice</b>:  A non well formed numeric value encountered in <b>/var/www/partkeepr/app/PartKeeprRequirements.php</b> on line <b>126</b><br />
{"success":true,"message":"Symfony2\/PartKeepr Requirements","warnings":["<b>a PHP accelerator should be installed<\/b><br\/>Install and\/or enable a <strong>PHP accelerator<\/strong> (highly recommended).<br\/>","<b>PHP APCu cache not found<\/b><br\/>For best performance install the PHP APCu cache<br\/>","<b>Maximum Execution Time might be too low<\/b><br\/>Your maximum execution time is set to 30 seconds, which might be too low for low-end systems. If you encounter problems, please increase the value.<br\/>"],"errors":[]}
log @ ext-all-debug.js:9392
```

**Comment by blakangel** [comment]
I fixed this problem by explicitly type casting the problem variable with (int) in PartKeeprRequirements.php as follows:
```
    protected function returnBytes($val)
    {
        $val = (int)trim($val);

```
I guess PHP's automatic type casting functionality has changed recently, but I'm unsure.

**Comment by christianlupus** [comment]
Just for reference: The linked image shows this:

![PBFh10L](https://user-images.githubusercontent.com/8202922/72440947-11c03a80-37aa-11ea-9953-adb8e5642ee4.png)

**Comment by christianlupus** [comment]
@vogelfreiheit @blakangel Can you verify if the issue is still present in the latest git version?

**Comment by baradhili** [comment]
@christianlupus this is no longer an issue for a fresh install from git I did it only a week ago

**Comment by christianlupus** [comment]
OK, will close and reopen if need arises.

**Comment by mkamarin** [comment]
I am having the same error. Using PartKeepr 1.4.0 (downloaded couple of weeks ago) on Linux Mint 20 with XAMPP (XAMPP for Linux 7.4.8 [PHP 7.4.8] https://www.apachefriends.org/index.html).

The php_error_log includes the following line:

[01-Aug-2020 17:50:45 America/Los_Angeles] PHP Notice:  A non well formed numeric value encountered in /xxxxxx/www/htdocs/partkeepr-1.4.0/app/PartKeeprRequirements.php on line 126

![Screenshot from 2020-08-01 17-55-28](https://user-images.githubusercontent.com/39018002/89113089-189a5c00-d421-11ea-969d-1cc2ddd04842.jpg)

The problem is in:
```php
    protected function returnBytes($val)
    {
        $val = trim($val);
        $last = strtolower($val[strlen($val) - 1]);
        switch ($last) {
            // The 'G' modifier is available since PHP 5.1.0
            case 'g':
                $val *= 1073741824;
                break;
            case 'm':
                $val *= 1048576; // <= line 126
                break;
            case 'k':
                $val *= 1024;
                break;
        }

        return $val;
    }
```

Tried casting the line as follows:
```
    $val *= (int)1048576; // <= line 126
```
but it does not solve the problem. Any suggestion will be welcome, as I don't  know what to do next.
